dashboard/src/actions/overviewActions.js

```diff
@@ -0,0 +1,208 @@
+        `${endpoints?.api?.datasets_metadata}/${dataset.resource_id}`,
+        {
+          metadata: { [method]: actionValue },
+        }
+      );
+      if (response.status === 200) {
+        if (actionType === "save") {
+          savedRuns.push(dataset);
+          const filteredNewRuns = newRuns.filter(
+            (item) => item.resource_id !== dataset.resource_id
+          );
+          const filteredInitRuns = initNewRuns.filter(
+            (item) => item.resource_id !== dataset.resource_id
+          );
+          dispatch({
+            type: TYPES.SAVED_RUNS,
+            payload: savedRuns,
+          });
+          dispatch({
+            type: TYPES.INIT_NEW_RUNS,
+            payload: filteredInitRuns,
+          });
+          dispatch({
+            type: TYPES.NEW_RUNS,
+            payload: filteredNewRuns,
+          });
+        } else {
+          const dataIndex = newRuns.findIndex(
+            (item) => item.resource_id === dataset.resource_id
+          );
+          newRuns[dataIndex].metadata[metaDataActions[actionType]] =
+            response.data[metaDataActions[actionType]];
+          dispatch({
+            type: TYPES.NEW_RUNS,
+            payload: newRuns,
+          });
+        }
+      } else {
+        dispatch(constructToast("danger", response?.data?.message));
+      }
+    } catch (error) {
+      dispatch(constructToast("danger", error?.response?.data?.message));
+      dispatch({ type: TYPES.NETWORK_ERROR });
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  };
+
+export const deleteDataset = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    const response = await API.post(
+      `${endpoints?.api?.datasets_delete}/${dataset.resource_id}`
+    );
+    if (response.status === 200) {
+      const datasets = getState().overview.newRuns;
+      const initNewRuns = getState().overview.initNewRuns;
+      const result = datasets.filter(
+        (item) => item.resource_id !== dataset.resource_id
+      );
+      const filteredInitRuns = initNewRuns.filter(
+        (item) => item.resource_id !== dataset.resource_id
+      );
+      // dispatch is awaited else toast message will appear before the UI updated
+      await dispatch({
+        type: TYPES.INIT_NEW_RUNS,
+        payload: filteredInitRuns,
+      });
+
+      await dispatch({
+        type: TYPES.NEW_RUNS,
+        payload: result,
+      });
+      dispatch(constructToast("success", "Deleted!"));
+    }
+  } catch (error) {
+    dispatch(constructToast("danger", error?.response?.data?.message));
+    dispatch({ type: TYPES.NETWORK_ERROR });
+  }
+  dispatch({ type: TYPES.COMPLETED });
+};
+
+export const setRows = (rows) => {
+  return {
+    type: TYPES.INIT_NEW_RUNS,
+    payload: rows,
+  };
+};
+
+export const setSelectedRuns = (rows) => {
+  return {
+    type: TYPES.SELECTED_NEW_RUNS,
+    payload: rows,
+  };
+};
```

Comment:
I don't think it's going to make much difference

Reply:
The basic ideas are are encapsulation and interface design.  That is, we want to export from a module only the symbols that other modules need; we want to import into a module only the symbols that the current module needs; and, within reason, we want the numbers to be as small as reasonable.  And, functional interfaces are generally more flexible, resilient, and easier to maintain over time than data-based ones.

So...component modules like `overviewActions` need to dispatch actions, and the question is, how should they do that.  The prevailing practice seems to be to use an idiom like:
```
    dispatch(actionCreationFunction());
```
Thus, the module making that call needs a definition of the action creation function which in turn needs a definition of the action type.  If we define the action creation function in the module making the call, then the module needs to import the action type definitions, and the action creation function cannot be readily shared with other component modules.  Instead, we could define the action creation function in the same module as the reducers that it triggers.  This means that the type definitions don't need to be exported at all:  the types are the means of communicating between the dispatched action and the reducer which executes it, and if the action creation is done in the same module as where it is executed, then the commonality can be encapsulated, and only the action creation functions interfaces need to be executed...which is a very desirable thing, in general.

At the very lowest level, you are right, it makes little difference -- the `overviewAction` module needs to call a function which needs to encode the type and the result needs to be dispatched to a reducer function -- but, from an organizational and interface perspective, it makes a fair bit of difference:  by encapsulating and hiding the type definitions, we create a cleaner, functional interface for the components to use which better decouples the components from the implementation of the actions.  The components don't have to be concerned with the names of the action types or their representations nor with the details of the action payloads; there is just a functional interface which they call whose result they pass to `dispatch()`.  This enables things like being able to change seamlessly between actions and thunks without modifying the caller, because the caller's interface is (potentially) the same for either one.

So, yeah, it's just a detail in how the code is organized, but those details can enable very powerful things in terms of code robustness and maintainability.